libvirt/tests/src/virtual_device/filesystem_device.py

```diff
@@ -56,31 +56,45 @@ def shared_data(vm_names, fs_devs):
             session = vm.wait_for_login()
             for fs_dev in fs_devs:
                 logging.debug(fs_dev)
-                session.cmd('rm -rf %s' % fs_dev.source['dir'], ignore_all_errors=False)
-                session.cmd('mkdir -p %s' % fs_dev.source['dir'])
+                mount_dir = '/var/tmp/' + fs_dev.target['dir']
+                session.cmd('rm -rf %s' % mount_dir, ignore_all_errors=False)
+                session.cmd('mkdir -p %s' % mount_dir)
                 logging.debug("mount virtiofs dir in guest")
-                cmd = "mount -t virtiofs %s %s" % (fs_dev.target['dir'], fs_dev.source['dir'])
+                cmd = "mount -t virtiofs %s %s" % (fs_dev.target['dir'], mount_dir)
                 status, output = session.cmd_status_output(cmd, timeout=300)
                 if status != 0:
                     session.close()
                     test.fail("mount virtiofs dir failed: %s" % output)
                 if vm == vms[0]:
-                    filename = fs_dev.source['dir'] + '/' + vm.name
-                    cmd = "dd if=/dev/urandom of=%s bs=1M count=512 oflag=direct" % filename
+                    filename_guest = mount_dir + '/' + vm.name
+                    #filename_host = fs_dev.source['dir'] + '/' +vm.name
+                    cmd = "dd if=/dev/urandom of=%s bs=1M count=512 oflag=direct" % filename_guest
                     status, output = session.cmd_status_output(cmd, timeout=300)
                     if status != 0:
                         session.close()
                         test.fail("Write data failed: %s" % output)
-                md5_value = session.cmd_status_output("md5sum %s" % filename)[1].strip()
+                md5_value = session.cmd_status_output("md5sum %s" % filename_guest)[1].strip().split()[0]
                 md5s.append(md5_value)
                 logging.debug(md5_value)
-                md5_value = process.run("md5sum %s" % filename).stdout_text.strip()
+                md5_value = process.run("md5sum %s" % filename_guest).stdout_text.strip().split()[0]
                 logging.debug(md5_value)
                 md5s.append(md5_value)
             session.close()
         if len(set(md5s)) != len(fs_devs):
             test.fail("The md5sum value are not the same in guests and host")
 
+    def launch_externally_virtiofs(source_dir, source_socket):
+        """
+        launch externally virtiofs:
```

Comment:
Capital 
```suggestion
        Launch externally virtiofs
```

I think we'd better capitalize the first letter and remove the ':' .

Reply:
Modified.